jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/CaseServiceImplTest.java

```diff
@@ -3730,4 +3735,48 @@ public void testCaseWithStageAndBoundaryTimerFired() throws InterruptedException
             }
         }
     }
+
+    @Test
+    public void testCaseWithBoundaryTimerFiredAtStage() throws InterruptedException {
+        ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+        PrintStream originalOut = System.out;
+
+        System.setOut(new PrintStream(outContent));
+
+        String caseId = caseService.startCase(deploymentUnit.getIdentifier(), "CaseWithBoundaryTimerStage");
+        assertNotNull(caseId);
+        assertEquals(FIRST_CASE_ID, caseId);
+        
+        Thread.sleep(3000);
```

Comment:
we should use event listeners with countdown latch instead of thread sleep. See other timer related tests.

Reply:
Thanks @mswiderski . 
Fixed at @elguardian fork for being merged there